runtime/interpreter/value.go

```diff
@@ -17907,7 +17920,10 @@ func (i DictionaryIterator) NextKey(gauge common.MemoryGauge) Value {
 }
 
 func (v *DictionaryValue) Iterator() DictionaryIterator {
-	mapIterator, err := v.dictionary.Iterator()
+	// TODO: Is it safe to use ReadOnly Iterator here
+	// because DictionaryIterator is only used to iterate keys?
+	// Maybe rename DictionaryIterator to DictionaryKeyIterator.
+	mapIterator, err := v.dictionary.ReadOnlyIterator()
```

Comment:
We need to check where `DictionaryValue.Iterator` is used. The key might still be used to perform a mutating operation. It might be better to be conservative here and assume mutation, i.e. reverting this  switch to a read-only iterator and leaving the TODO, we can optimize later

Reply:
Make sense.  However, it seems we may need to modify `Iterator()` API to add these new parameters for non-readonly iterator:

```Go
valueComparator := newValueComparator(interpreter, EmptyLocationRange)
hashInputProvider := newHashInputProvider(interpreter, EmptyLocationRange)
```

Current `Iterator` API:
```Go
func (v *DictionaryValue) Iterator() DictionaryIterator {
```

Reply:
Yeah, the signature of `DictionaryValue.Iterator` should be changed to include the `*Interpreter` and the `LocationRange`.

Reply:
@fxamacker As you mentioned above, but I didn't understand at the time, `DictionaryIterator` is only used for key iteration at the moment. Refactored the naming and removed the TODO in a93c048de48505a6e09a004f3487042233d6d172